include/exadg/postprocessor/write_output.h

```diff
@@ -78,13 +81,60 @@ write_boundary_IDs(dealii::Triangulation<dim> const & triangulation,
 template<int dim>
 void
 write_grid(dealii::Triangulation<dim> const & triangulation,
+           dealii::Mapping<dim> const &       mapping,
+           unsigned int const                 n_subdivisions,
            std::string const &                folder,
-           std::string const &                file)
+           std::string const &                file,
+           unsigned int const &               counter,
+           MPI_Comm const &                   mpi_comm)
 {
-  std::string filename = folder + file + "_grid";
+  std::string filename = file + "_grid";
 
-  dealii::GridOut grid_out;
-  grid_out.write_mesh_per_processor_as_vtu(triangulation, filename);
+  dealii::DataOut<dim> data_out;
+
+  dealii::DataOutBase::VtkFlags flags;
+  flags.write_higher_order_cells = n_subdivisions > 1;
+  data_out.set_flags(flags);
+
+  data_out.attach_triangulation(triangulation);
+  data_out.build_patches(mapping, 4, dealii::DataOut<dim>::curved_inner_cells);
+  data_out.write_vtu_with_pvtu_record(folder, filename, counter, mpi_comm, 4);
+}
+
+template<int dim>
+void
+write_points(std::vector<dealii::Point<dim>> const & points,
+             std::string const &                     folder,
+             std::string const &                     file,
+             unsigned int const &                    counter,
+             MPI_Comm const &                        mpi_comm)
+{
+  std::string filename = file + "_points";
+
+  double min_coord = points[0][0];
+  double max_coord = points[0][0];
+  for(auto const & p : points)
+  {
+    for(unsigned int d = 0; d < dim; ++d)
+    {
+      min_coord = std::min(p[d], min_coord);
+      max_coord = std::max(p[d], max_coord);
+    }
+  }
+  min_coord = dealii::Utilities::MPI::min(min_coord, mpi_comm);
+  max_coord = dealii::Utilities::MPI::max(max_coord, mpi_comm);
```

Comment:
This looks like some generic functionality that could be realized independently. I wonder whether this is already available in deal.II as some (particle) utility function? @peterrum @kronbichler Do you know something? If not, I suggest to go with the present code version.

Reply:
If one wants to provide min/max as utility functions, I think one should then provide independent functions for min and for max. One could also think of some norms or min/max values of `dealii::Point`. Then, one only needs to loop over the vector (plus MPI afterwards).